03week/towersOfHanoi.js

```diff
@@ -13,33 +13,129 @@ let stacks = {
   c: []
 };
 
+// function to sanitize user input
+// returns a lower case, trimmed string
+function sanitizeInput(input) {
+  return input.toLowerCase().trim();
+}
+
+// validates user input
+// returns true/false
+// TRUE: if input is: a, b, c
+// FALSE: otherwise
+function validateInput(startStack, endStack) {
+  const start = sanitizeInput(startStack);
+  const end = sanitizeInput(endStack);
+
+  return ((start === 'a' || start === 'b' || start === 'c') && (end === 'a' || end === 'b' || end === 'c'));
+}
+
+// prints the current stacks object
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// moves a piece from one stack to the other
+// will only move piece if the move isLegal
+// does not return anything
+function movePiece(startStack, endStack) {
+  // sanitize user input
+  const start = sanitizeInput(startStack);
+  const end = sanitizeInput(endStack);
+
+  // if is a legal move, then move piece
+  if ( isLegal(start, end) ) {
```

Comment:
isLegal shouldn't be called in movePiece, it should be called in the parent function. You're move piece function shouldn't check if the move is legal, it should only move the piece